File: boutique_back/src/main/java/com/boutique/entity/Pago.java
package com.boutique.entity;

import com.boutique.entity.enums.EstadoPago;
import com.boutique.entity.enums.MetodoPago;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalTime;

/**
 * @author GERSON
 */

@Getter
@Setter
@ToString(exclude = {"venta", "cuota"})
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
public class Pago {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private LocalDate fecha;

    @Column(nullable = false)
    private LocalTime hora;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, name = "metodo_pago", length = 20)
    private MetodoPago metodoPago;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal monto;

    @Column(name = "pago_de", length = 20)
    private String pagoDe;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private EstadoPago estado;

    // Dirección bidireccional con Venta
    @OneToOne(mappedBy = "pago")
    private Venta venta;

    @OneToOne(mappedBy = "pago")
    private Cuota cuota;
}

File: boutique_front/src/types/pago.ts
import type { Cuota } from "./cuota";
import type { Venta } from "./venta";

export enum MetodoPago {
    EFECTIVO = 'EFECTIVO',
    TARJETA = 'TARJETA',
    QR = 'QR',
}

export enum EstadoPago {
    PENDIENTE = 'PENDIENTE',
    COMPLETADO = 'COMPLETADO',
    FALLIDO = 'FALLIDO',
}

export interface Pago {
    id: number | null;
    fecha: string;
    hora: string;
    metodoPago: MetodoPago;
    monto: number;
    pagoDe: string | null;
    estado: EstadoPago | null;
    idVenta: number | null;
    idCuota: number | null;
}

export interface PagoDetalle {
    id: number;
    fecha: string;
    hora: string;
    metodoPago: MetodoPago;
    monto: number;
    pagoDe: string;
    estado: EstadoPago;
    venta: Venta | null;
    cuota: Cuota | null;
}
